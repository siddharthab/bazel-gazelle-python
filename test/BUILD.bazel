load("@bazel_gazelle//:def.bzl", "gazelle_binary", "gazelle_generation_test")
load("@gazelle_python//manifest:defs.bzl", "python_external_modules_manifest")
load("@pip//:requirements.bzl", "all_whl_requirements")
load("@rules_python//python:pip.bzl", "compile_pip_requirements")

compile_pip_requirements(
    name = "requirements",
    extra_args = [
        "--allow-unsafe",
        "--resolver=backtracking",
    ],
    requirements_in = "requirements.in",
    requirements_txt = "requirements.txt",
)

python_external_modules_manifest(
    name = "external_modules",
    exclude_patterns = [
        "^_|(\\._)+",
    ],
    wheels = all_whl_requirements,
)

gazelle_binary(
    name = "gazelle_binary",
    languages = [
        "@gazelle_python//python",
    ],
)

gazelle_generation_test(
    name = "gazelle_test",
    gazelle_binary = ":gazelle_binary",
    test_data = glob(
        include = ["**"],
        exclude = [
            "*",
            "bazel-*/**",
        ],
    ),
)
