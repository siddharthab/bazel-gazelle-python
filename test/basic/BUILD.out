# gazelle:py_root_dir python
# gazelle:py_repo_name_prefix pip_
# gazelle:py_internal_module_list_path internal_modules.txt
# gazelle:py_external_module_map_path external_modules.tsv
